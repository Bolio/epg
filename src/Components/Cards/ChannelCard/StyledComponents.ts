import styled from "styled-components";

export const StyledChannelCard = styled.div`
  width: 100%;
  height: 100px;
  padding: 0.55rem;
  background-color: #1a1a1a;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border: solid 1px black;
  border-right: solid 6px black;

  div.channel-number {
    p {
      margin: 0;
      font-size: 1.25rem;
      font-weight: 600;
    }
  }

  div.image-container {
    width: 150px;

    img {
      width: 100%;
    }
  }

  @media screen and (min-width: 768px) {
    div.channel-number {
      p {
        font-size: 2rem;
      }
    }
  }
`;
