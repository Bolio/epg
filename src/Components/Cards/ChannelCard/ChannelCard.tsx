import { StyledChannelCard } from "./StyledComponents";

interface IProps {
  number: number;
  image: string;
}

const ChannelCard = ({ number, image }: IProps) => {
  return (
    <StyledChannelCard>
      <div className="channel-number">
        <p>{number}</p>
      </div>
      <div className="image-container">
        <img src={image} />
      </div>
    </StyledChannelCard>
  );
};

export default ChannelCard;
