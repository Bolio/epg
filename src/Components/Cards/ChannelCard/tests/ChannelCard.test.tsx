import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ChannelCard from "../ChannelCard";

describe("ChannelCard component", () => {
  it("Should render the channel number", () => {
    const number = 123;
    const image = "test-image.jpg";
    const { getByText } = render(<ChannelCard number={number} image={image} />);
    expect(getByText(number.toString())).toBeInTheDocument();
  });

  it("Should render the image", () => {
    const number = 123;
    const image = "test-image.jpg";
    const { getByRole } = render(<ChannelCard number={number} image={image} />);
    const imgElement = getByRole("img");
    expect(imgElement).toHaveAttribute("src", image);
  });
});
