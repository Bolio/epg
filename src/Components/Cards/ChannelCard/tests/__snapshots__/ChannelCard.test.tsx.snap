// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelCard component Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh bCfLgt"
  >
    <div
      class="channel-number"
    >
      <p>
        123
      </p>
    </div>
    <div
      class="image-container"
    >
      <img
        src="test-image.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
